checks.d/spark.py

```diff
@@ -0,0 +1,444 @@
+
+        # Get the executor metrics
+        self._spark_executor_metrics(spark_apps, tags)
+
+        # Get the rdd metrics
+        self._spark_rdd_metrics(spark_apps, tags)
+
+        # Report success after gathering all metrics from the ApplicationMaster
+        if running_apps:
+            app_id, (app_name, tracking_url) = running_apps.items()[0]
+            am_address = self._get_url_base(tracking_url)
+
+            self.service_check(SPARK_SERVICE_CHECK,
+                AgentCheck.OK,
+                tags=['url:%s' % am_address],
+                message='Connection to ApplicationMaster "%s" was successful' % rm_address)
+
+    def _get_running_spark_apps(self, rm_address):
+        '''
+        Return a dictionary of {app_id: (app_name, tracking_url)} for the running Spark applications
+        '''
+        metrics_json = self._rest_request_to_json(rm_address,
+            YARN_APPS_PATH,
+            YARN_SERVICE_CHECK,
+            states=YARN_APPLICATION_STATES,
+            applicationTypes=YARN_APPLICATION_TYPES)
+
+        running_apps = {}
+
+        if metrics_json.get('apps'):
+            if metrics_json['apps'].get('app') is not None:
+
+                for app_json in metrics_json['apps']['app']:
+                    app_id = app_json.get('id')
+                    tracking_url = app_json.get('trackingUrl')
+                    app_name = app_json.get('name')
+
+                    if app_id and tracking_url and app_name:
+                        running_apps[app_id] = (app_name, tracking_url)
+
+        return running_apps
+
+    def _get_spark_app_ids(self, running_apps):
+        '''
+        Return a dictionary of {app_id: (app_name, tracking_url)} for Spark applications
+        '''
+        spark_apps = {}
+        for app_id, (app_name, tracking_url) in running_apps.iteritems():
+            response = self._rest_request_to_json(tracking_url,
+                SPARK_APPS_PATH,
+                SPARK_SERVICE_CHECK)
+            for app in response:
+                app_id = app.get('id')
+                app_name = app.get('name')
+
+                if app_id and app_name:
+                    spark_apps[app_id] = (app_name, tracking_url)
+
+        return spark_apps
+
+    def _spark_job_metrics(self, running_apps, addl_tags):
+        '''
+        Get metrics for each Spark job.
+        Return a map from Stage IDs to Job IDs
+        '''
+        for app_id, (app_name, tracking_url) in running_apps.iteritems():
+
+            response = self._rest_request_to_json(tracking_url,
+                SPARK_APPS_PATH,
+                SPARK_SERVICE_CHECK, app_id, 'jobs')
+
+            tags = ['app_name:' + str(app_name)]
+            tags.extend(addl_tags)
+
+            for job in response:
+                self._set_metrics_from_json(tags, job, SPARK_JOB_METRICS)
+
+            if len(response):
+                self._set_metric('spark.job.count', GAUGE, len(response), tags)
```

Comment:
We'll get into the issue of the gauges overwriting one another if multiple apps have the same `app_name`. I'd recommend using `increment` instead of `gauge` (see https://github.com/DataDog/dd-agent/blob/5.7.3/checks/__init__.py#L364) for the `spark.*.count` metrics that you've added.

Here, you can simply call `self.increment` for each app with the number of jobs of that app, and the metric that's ultimately sent would contain the sum of the numbers that have been passed to `increment`.

So here, if multiple apps share the same `app_name`, the metric sent for that `app_name` would be the total number of jobs over these apps, which sounds okay to me.